cloudflare/resource_cloudflare_worker_secret.go

```diff
@@ -0,0 +1,119 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareWorkerSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareWorkerSecretCreate,
+		Read:   resourceCloudflareWorkerSecretRead,
+		Delete: resourceCloudflareWorkerSecretDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareWorkersSecretImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"script_name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The name of the Worker script to associate the secret with.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The name of the Worker secret.",
+			},
+			"secret_text": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Description: "The text of the Worker secret, this cannot be read back after creation and is stored encrypted .",
+			},
+		},
+	}
+}
+
+func resourceCloudflareWorkerSecretRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	scriptName := d.Get("script_name").(string)
+	name := d.Get("name").(string)
+
+	secrets, err := client.ListWorkersSecrets(context.Background(), scriptName)
+	if err != nil {
+		return errors.Wrap(err, "error reading worker secrets")
+	}
+
+	for _, secret := range secrets.Result {
+		if secret.Name == name {
+			return nil
+		}
+	}
+
+	d.SetId("")
```

Comment:
calling [`d.SetId("")` removes the resource from the state](https://github.com/hashicorp/terraform/blob/master/helper/schema/resource_data.go#L255-L272) which i don't think you want here. typically in the `Read` operation, we sync all the values from the remote call to ensure we have the latest values. e.g.

```golang
d.Set("script_name", secrets.ScriptName)
d.Set("name", secrets.Name)
d.SetId(stringChecksum(....)) // maybe combine the name + script name?
```

Reply:
Sure, the value cannot be retrieved however it should still be persisted to state. Without it, every Terraform operation will re-create this.

Reply:
I've decided that the value must always be recreated as the existing secret cannot be checked against what is being presented to the resource.

Reply:
Unfortunately as there is no way to retrieve the secret values from Cloudflare, the drift cannot be detected, and so the secrets must be recreated/overwritten each time the plan is executed. This is as per the resource provider for the Hashicorp Vault provider: https://github.com/hashicorp/terraform-provider-vault/blob/master/vault/resource_generic_secret.go